internal/tofu/context_plan2_test.go

```diff
@@ -4718,92 +4883,230 @@ import {
 	}
 }
 
-func TestContext2Plan_importToInvalidDynamicAddress(t *testing.T) {
+func TestContext2Plan_importWithInvalidForEach(t *testing.T) {
 	type TestConfiguration struct {
 		Description         string
 		expectedError       string
 		inlineConfiguration map[string]string
 	}
 	configurations := []TestConfiguration{
 		{
-			Description:   "To address index value is null",
-			expectedError: "Import block 'to' address contains an invalid key: Import block contained a resource address using an index which is null. Please ensure the expression for the index is not null",
+			Description:   "for_each value is null",
+			expectedError: "Invalid import id argument: The import ID cannot be null",
 			inlineConfiguration: map[string]string{
 				"main.tf": `
-variable "index" {
-  default = null
+locals {
+  map = {
+    "key1" = null
+  }
+}
+
+resource "test_object" "a" {
+  for_each = local.map
+}
+
+import {
+  for_each = local.map
+  to = test_object.a[each.key]
+  id = each.value
+}
+`,
+			},
+		},
+		{
+			Description:   "for_each key is null",
+			expectedError: "Null value as key: Can't use a null value as a key.",
+			inlineConfiguration: map[string]string{
+				"main.tf": `
+variable "nil" {
+	default = null
+}
+
+locals {
+  map = {
+    (var.nil) = "val1"
+  }
+}
+
+resource "test_object" "a" {
+  for_each = local.map
+}
+
+import {
+  for_each = local.map
+  to = test_object.a[each.key]
+  id = each.value
+}
+`,
+			},
+		},
+		{
+			Description:   "for_each expression is null",
+			expectedError: "Invalid for_each argument: The given \"for_each\" argument value is unsuitable: the given \"for_each\" argument value is null. A map, set of strings, or a tuple is allowed.",
```

Comment:
Just a minor readability comment: we could use backticks for strings that contain double quotes to not escape them.

Like this: \`Invalid for_each argument: The given "for_each" argument value is unsuitable: the given "for_each" argument value is null. A map, set of strings, or a tuple is allowed.\`

Reply:
Fixed https://github.com/opentofu/opentofu/pull/1645/commits/53a62c802bb61cb68963a2727a2fee590fc37cf3